### DISCUSSION TITLE
theme integration

### DISCUSSION THREAD

**Discussion by BKSalman** [hecrj/iced#1470] [Q&A] [answered]
Hello, I was wondering how would I go about integrating the application's theme with the host pc theme, 

for example I'm using KDE on linux, and I use a custom dark theme, and I want the application to use the same theme,

how would that work?

**Comment by felipetesc** [comment, marked as answer]
Your question is really interesting. I also want to know this. So, I googled and the answer, though, isn't that nice. There is a clarifying post at https://stackoverflow.com/questions/16254895/x11-how-to-get-system-colors .  There you will find:  _"It could be all so simple, weren't both Qt and GTK+/GNOME serious cases of the NIH syndrome.."_, as explained by [datenwolf](https://stackoverflow.com/users/524368/datenwolf). 


You would need to read the theme's configuration files from your custom theme, If this functionality is only for KDE, than another nice post can be found at [Get system colors from KDE](https://stackoverflow.com/questions/17640514/get-system-colors-from-kde).  Once you have those files,  they could to be serialized into a structure called OSTheme, for example. In my sys,  I can find these files inside _**/home/user_name/.kde4/share/apps/color-schemes/**_, the location may vary.

After that you you need to convert KDE values to Ice's values

So: 
- find the conf files inside your sys
- read the files
- serialize
- convert serialized values from KDE to Ice theme
- and finally access the KDEToIced values inside your app

Other nice readings:
- https://docs.kde.org/stable5/en/kate/katepart/color-themes.html
- https://forum.kde.org/viewtopic.php?t=132722
- https://askubuntu.com/questions/568185/is-kde5-backwards-compatible-with-apps-widgets-themes-from-kde4

**Reply by BKSalman** [reply]
IIIIIIIIIIIIIINTERESTINGGGGGGGGGGG!!

I might try to make a prototype later that could use the system's theme, thanks for spending the time to research this and share your findings!!

**Reply by felipetesc** [reply]
It's always a pleasure to help.
